src/db/spanner/models.rs

```diff
@@ -221,13 +255,27 @@ impl SpannerDb {
         Ok(id)
     }
 
+
+    #[allow(dead_code)]
     fn get_or_create_collection_id(&self, name: &str) -> Result<i32> {
         self.get_collection_id(name).or_else(|e| match e.kind() {
             DbErrorKind::CollectionNotFound => self.create_collection(name),
             _ => Err(e),
         })
     }
 
+    async fn get_or_create_collection_id_async(&self, name: &str) -> Result<i32> {
+        let result = self.get_collection_id_async(name).await;
```

Comment:
can we keep the original `or_else` or is that now troublesome here?

Reply:
It's troublesome because async closures aren't allowed yet.